File: scripts/corpus-statistics.js
const fs = require('fs');
const path = require('path');

// Set the directory path
const directoryPath = '../atas';

// Function that calculate the metrics
function calculateMetrics(text) {
    // Count number of words
    const words = text.split(/\s+/);
    const wordCount = words.length;

    // Count number of characters with spaces
    const charCountWithSpaces = text.length;

    // Count number of characters without spaces
    const charCountWithoutSpaces = text.replace(/\s+/g, '').length;

    // Count number of sentences
    const sentences = text.split(/[.?!]+/);
    const sentenceCount = sentences.length - 1;

    // Calculate other metrics
    const charsPerWords = charCountWithoutSpaces / wordCount;
    const wordsPerSentence = wordCount / sentenceCount;

    // Return an object containing the calculated metrics
    return {
        charCountWithSpaces,
        charCountWithoutSpaces,
        wordCount,
        sentenceCount,
        charsPerWords,
        wordsPerSentence,
    };
}


// Read all files in the directory
fs.readdir(directoryPath, function (err, files) {
    if (err) {
        console.log('Error reading directory:', err);
        return;
    }

    // Filter the files to only include text files
    const textFiles = files.filter(function (file) {
        return path.extname(file).toLowerCase() === '.txt';
    });

    // Read the contents of each text file and store them in an array
    const contents = [];
    textFiles.forEach(function (file) {
        const filePath = path.join(directoryPath, file);
        const fileContents = fs.readFileSync(filePath, 'utf8');
        contents.push(fileContents);
    });

    console.log(contents.length, "atas");
    const corpus = contents.join(' ');
    const metrics = calculateMetrics(corpus);
    console.log(metrics);

});

File: scripts/corpus-statistics.jl
using WordTokenizers

listAtas = readdir("../atas");
corpus = [];

for i in eachindex(listAtas)
    f = open("../atas/" * listAtas[i])
    txt = readlines(f)
    push!(corpus, txt)
end
println(length(corpus), " atas")

corpus = join(corpus, " ");

function text_stats(text::AbstractString)
    num_words = length(split(text))
    num_chars_with_spaces = length(text)
    num_chars_without_spaces = length(replace(text, " " => ""))
    num_sentences = length(split_sentences(text))
    num_chars_per_word = num_chars_without_spaces / num_words
    num_words_per_sentence = num_words / num_sentences

    return num_words, num_chars_with_spaces, num_chars_without_spaces, num_sentences, num_chars_per_word, num_words_per_sentence
end

num_words, num_chars_with_spaces, num_chars_without_spaces, num_sentences, num_chars_per_word, num_words_per_sentence = text_stats(corpus)
println("Number of characters (with spaces): ", num_chars_with_spaces)
println("Number of characters (without spaces): ", num_chars_without_spaces)
println("Number of words: ", num_words)
println("Number of sentences: ", num_sentences)
println("Number of characters per words: ", num_chars_per_word)
println("Number of words per sentence: ", num_words_per_sentence)